[Host_v2] - Question and Answer Cards

Alexandra-Kushnirsky opened the pull request:
**Issue:**
Create styling for question and answer cards in host version 2. Corresponding git issue with figma here: https://github.com/rightoneducation/righton-app/issues/913  


**Resolution:**

Added:
(in `host_v2>src>components`)
- new component for question card: `QuestionCard.tsx` 
- corresponding story for the above: `QuestionCard.stories.tsx` 
- new component for answer card: `AnswerCard.tsx` 
- corresponding story for the above: `AnswerCard.stories.tsx` 

(in `host_v2>src>lib>styledcomponents`)
- Added new styled component card for answer option: `AnswerOptionStyled.tsx`

(in `host_v2>src>lib>Theme.tsx`)
- Added constant border radius value for answer options on answer cards: `lines 53, 69, 85, 144`

(in `host_v2>public>locales`)
- Added Question card title to dictionary in `en>translation.json`
- Added Question card title to dictionary in `es>translation.json`

(in `host_v2>src>components>PlaceholderContentArea.tsx`)
- Added new question and answer card components to placeholder content for testing purposes



**Preview:**
Here are some screenshots of the new cards (1) incorrect answer card (2) correct answer card:
<img width="978" alt="Screenshot 2023-12-28 at 12 40 12 PM" src="https://github.com/rightoneducation/righton-app/assets/106450593/70344a4b-b02a-4d04-bd5b-97b6d227867c">

<img width="937" alt="Screenshot 2023-12-28 at 12 40 52 PM" src="https://github.com/rightoneducation/righton-app/assets/106450593/e0830343-c2ed-4339-9d88-6491eb47a580">

Here is a link to the storybook as well: https://64220218d457f84ba4546eb1-bhtlqubulm.chromatic.com/?path=/story/design-system-3-organisms-card @amarax 

@drewjhart

drewjhart commented on host_v2/src/lib/styledcomponents/AnswerOptionStyled.tsx:
```diff
@@ -0,0 +1,10 @@
+import { styled, useTheme } from '@mui/material/styles';
```
nice job passing the theme through on this

drewjhart commented on host_v2/src/components/AnswerCard.tsx:
```diff
@@ -0,0 +1,90 @@
+import BodyCardStyled from '../lib/styledcomponents/BodyCardStyled';
+import AnswerOptionStyled from '../lib/styledcomponents/AnswerOptionStyled';
+
+interface AnswerCardProps {
```
Given that some of these values might be null in some cases, I think our interface should reflect that? Like instructions: string[] | null or whatever values are sometimes not going to be provided to the component.

drewjhart reviewed (changes requested):
I think just the interface needs to be updated (per one of my comments). The merge conflict github is identifying is a nonissue (you can just accept your code)

Alexandra-Kushnirsky commented:
Thanks again for helping me work through the git issues! Just so you know, I made a few more changes (in addition to addressing your comments) per what we discussed in pair programming today. Specifically, I added an answer option interface (with all the fields previously being passed in: instructions, reason, content, isCorrectAnswer) for each answer option and a question data interface (with imageUrl and questionText). I know we said we will probably handle this some other way in the future (i.e. not creating objects for storing data fields) but just thought this might be better for the time being so as not to clutter the grandparent component with hard-coded sample data
